packages/shared/friends/sagas.ts

```diff
@@ -1179,3 +1193,241 @@ function updateSocialInfo(socialData: SocialData) {
     })
   )
 }
+
+// Join or create channel
+async function* handleJoinOrCreateChannel(action: JoinOrCreateChannel) {
```

Comment:
Sagas can't be async
```suggestion
function* handleJoinOrCreateChannel(action: JoinOrCreateChannel) {
```